backend/llm/composite_brain_qa.py

```diff
@@ -0,0 +1,611 @@
+            return self.generate_answer
+        elif brain.brain_type == BrainType.API:
+            return APIBrainQA(
+                brain_id=brain.id,
+                chat_id=self.chat_id,
+                model=self.model,
+                max_tokens=self.max_tokens,
+                temperature=self.temperature,
+                streaming=self.streaming,
+                prompt_id=self.prompt_id,
+            ).generate_answer
+        elif brain.brain_type == BrainType.DOC:
+            return KnowledgeBrainQA(
+                brain_id=str(brain.id),
+                chat_id=self.chat_id,
+                model=self.model,
+                max_tokens=self.max_tokens,
+                temperature=self.temperature,
+                streaming=self.streaming,
+                prompt_id=self.prompt_id,
+            ).generate_answer
+
+    def get_stream_answer_generator_from_brain_type(self, brain: BrainEntity):
+        if brain.brain_type == BrainType.COMPOSITE:
+            return self.generate_stream
+        elif brain.brain_type == BrainType.API:
+            return APIBrainQA(
+                brain_id=brain.id,
+                chat_id=self.chat_id,
+                model=self.model,
+                max_tokens=self.max_tokens,
+                temperature=self.temperature,
+                streaming=self.streaming,
+                prompt_id=self.prompt_id,
+            ).generate_stream
+        elif brain.brain_type == BrainType.DOC:
+            return KnowledgeBrainQA(
+                brain_id=str(brain.id),
+                chat_id=self.chat_id,
+                model=self.model,
+                max_tokens=self.max_tokens,
+                temperature=self.temperature,
+                streaming=self.streaming,
+                prompt_id=self.prompt_id,
+            ).generate_stream
+
+    def generate_answer(
+        self, chat_id: UUID, question: ChatQuestion, save_answer: bool
+    ) -> str:
+        connected_brains = brain_service.get_connected_brains(self.brain_id)
+        if not connected_brains:
+            response = HeadlessQA(
+                chat_id=chat_id,
+                model=self.model,
+                max_tokens=self.max_tokens,
+                temperature=self.temperature,
+                streaming=self.streaming,
+                prompt_id=self.prompt_id,
+            ).generate_answer(chat_id, question)
+            brain = brain_service.get_brain_by_id(self.brain_id)
+            if save_answer:
+                new_chat = chat_service.update_chat_history(
+                    CreateChatHistory(
+                        **{
+                            "chat_id": chat_id,
+                            "user_message": question.question,
+                            "assistant": response.assistant,
+                            "brain_id": question.brain_id,
+                            "prompt_id": self.prompt_to_use_id,
+                        }
+                    )
+                )
```

Comment:
Maybe directly calling generate_answer of `HeadlessQA` with `save_answer = true` ?

Reply:
Something like :
```python

    def generate_answer(
        self, chat_id: UUID, question: ChatQuestion, save_answer: bool
    ) -> str:
        connected_brains = brain_service.get_connected_brains(self.brain_id)
        if not connected_brains:
            return HeadlessQA(
                chat_id=chat_id,
                model=self.model,
                max_tokens=self.max_tokens,
                temperature=self.temperature,
                streaming=self.streaming,
                prompt_id=self.prompt_id,
            ).generate_answer(chat_id, question, save_answer=save_answer)
```